### PR TITLE
Detect better the endianess of a build

### PR DESCRIPTION
**Pull request by montjoie**
The current way to detect endianess is to check if a BIG ENDIAN defconfig frag was added.
But this is wrong for defconfig which is are nativly big endian like arm
ixp4xx_defconfig.

So the ultimate way to know the endianess is to download the kernel
config and grep it.

Signed-off-by: Corentin Labbe <clabbe@baylibre.com>

### CODE REVIEW

**Comment by gctucker** [comment]
I think this should be done at build time, when the `.config` file has been generated.  Then it can be grepped, or we can use the existing method in the build steps to check if a config is enabled, and populate some meta-data flag with the result.  For example, a similar thing is done to see if modules or device tree is enabled.

File: kernelci/test.py [hunk 1]
```diff
@@ -108,7 +108,12 @@ def get_params(meta, target, plan_config, storage):
     rootfs = plan_config.rootfs
     defconfig_full = kernel['defconfig_full']
     defconfig = ''.join(defconfig_full.split('+')[:1])
-    endian = 'big' if 'BIG_ENDIAN' in defconfig_full else 'little'
+
+    if "endian" in kernel:
+        endian = kernel["endian"]
+    else:
```
Comment (gctucker) [inline kernelci/test.py:114]: I don't think we need to keep backward compatibility, the code is expecting the data to be valid so it should always have the `endian` attribute.

File: kernelci/build.py [hunk 2]
```diff
@@ -1173,6 +1173,11 @@ def run(self, jopt=None, verbose=False, opts=None):
         if target:
             kbmeta['image'] = target
 
+        if self._kernel_config_enabled('CPU_BIG_ENDIAN'):
```
Comment (gctucker) [inline kernelci/build.py:1176]: I think `endianness` is a bit more appropriate.

**Review by gctucker — COMMENTED** [review]
LGTM, just a couple of minor things.

File: kernelci/test.py (hunk 1 above)
Comment (montjoie) [reply kernelci/test.py:114]: what happen if someone use kci_test against an old build ? I think we should keep compatibility for this case.

**Comment by gctucker** [comment]
LGTM, let's check results on staging.

**Pull request merged**